### ISSUE TITLE
DELETE doesn't propagate the body

### ISSUE THREAD

**Issue by slinkydeveloper** [fabrikt-io/fabrikt#225]
The HTTP DELETE method supports a body, as defined in https://www.rfc-editor.org/rfc/rfc7231#section-4.3.5:

>    A payload within a DELETE request message has no defined semantics;
   sending a payload body on a DELETE request might cause some existing
   implementations to reject the request.

But the code generator simply ignores it. It should be easy to fix by changing this line of code: https://github.com/cjbooms/fabrikt/blob/master/src/main/kotlin/com/cjbooms/fabrikt/generators/client/OkHttpSimpleClientGenerator.kt#L212

**Comment by slinkydeveloper** [comment]
A proposed PR here: https://github.com/cjbooms/fabrikt/pull/226

**Comment by aderington** [comment]
I've also run into this issue.  I was curious if there was an plan to merge the posted PR (or similar). Thank you!

**Comment by cjbooms** [comment]
Open to merging a fix. From what i recall, it wasn't clear what that PR was doing, and requests for clarification or tests were ignored.

**Comment by aderington** [comment]
I agree that the current PR could use some tweaks.  I added a couple of comments to the PR with what I think is going on and a suggestion for improving the PR.  I have not cloned the repo though, and I'm definitely not an expert in code generation, so take it with a grain of salt.

tl;dr;

```
"DELETE" -> {
  val hasBodyParameters = parameters.any { it is BodyParameter }
  if (hasBodyParameters) {
      this.addRequestSerializerStatement("delete")
  } else {
       this.add("\n.delete()")
  }
}
```

**Comment by mrhoeve** [comment]
I checked the current implementation and this issue still appears to be present. The OkHttp client generator always generates `.delete()` for a DELETE operation, even when the OpenAPI operation defines a request body.

I also reviewed the earlier attempt in #226. It appears to have been closed because the contributor did not follow up with the requested clarification and tests, rather than because the requested behaviour was rejected. I would use the review feedback from that PR and make the body/no-body distinction directly in the DELETE request generation.

My proposed behaviour would be:

- A DELETE operation without a request body continues to generate `.delete()`, preserving the current output.
- A DELETE operation with a regular request body uses the existing request serialization path and generates `.delete(requestBody)`.
- A DELETE operation with a multipart request body generates `.delete(multipartBody)`.
- Other HTTP methods and existing specifications remain unaffected.

Although a DELETE request body has no generally defined HTTP semantics, an explicitly declared OpenAPI `requestBody` provides a clear indication that the described endpoint expects and supports one. The generator should therefore not silently discard it.

I would add focused tests covering:

- DELETE without a request body;
- DELETE with a JSON request body;
- DELETE with a multipart request body;
- the generated client code compiling successfully;
- unchanged output for existing specifications.

@cjbooms, does this sound like the intended solution? If so, I would be happy to prepare the fix and tests.
